front: drop editorLayers

clarani opened the pull request:
Layers settings were duplicated in the infrastructure editor, with editorLayers
and mapSettings.layersSettings.

We keep only layersSettings as it is already used in the rest of the application.

:warning: check that the display of level crossings still works correctly

clarani committed:
front: drop editorLayers

Layers settings were duplicated in the infrastructure editor, with editorLayers
and mapSettings.layersSettings.

We keep only layersSettings as it is already used in the rest of the application.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: get data from extended context instead of from the store

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: drop Layers in EditorLayersModal

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

sim51 commented:
The error layer on the infra is not working.
NB: I need to check if I enabled the error layer on the infra editor, then if I switch to the map, it is not enabled.

sim51 reviewed (commented):
Well done!

I checked the code, and I have nothing to say on it.
I tested it and it works great, except the for the error layer (I also tested the level crossings  layer). 

I need to test it bit more, but so far, so good.

Thanks @clarani

Math-R commented on front/public/locales/en/translation.json:
```diff
@@ -135,14 +135,13 @@
       "speed_sections": "Speed limits",
       "switches": "Switches",
-      "track_sections": "Track sections"
+      "tvds": "Track sections"
```
i think tvds is very unclear why did you changed it ?

clarani replied:
It's to simplify, since the key in layersSettings is not track_sections but tvds :/ I totally agree that it is very unclear `tvds`, it's only temporay, I will rename it in another PR :+1:

Math-R commented on front/src/applications/editor/tools/utils.ts:
```diff
@@ -34,6 +34,8 @@ import type { AppDispatch } from 'store';
 import type { Bbox } from 'types';
 import { nearestPointOnLine } from 'utils/geometry';
 
+import { getLayerSettingNameFromEditorLayer } from '../utils';
```
nit: i don't know why but i feel a bit confuse about importing utils in utils, looks like a bit confusing to me

clarani replied:
Yes I agree, but can we leave it for another PR ? I think it would require to re-think the architecture of the folders and utils

clarani committed:
front: update defaultMapSettings to always display the track sections

This change will only impact the editor application since the track section layer is automatically
activated in all the others applications.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani commented:
@sim51 the error layer is fixed !

sim51 reviewed (approved):
Error layer is working now, thanks @clarani 
I also created a tiny infra with the editor, and it works.
Good for me.
